firebase-ai/src/main/kotlin/com/google/firebase/ai/type/Tool.kt

```diff
@@ -43,7 +51,26 @@ internal constructor(internal val functionDeclarations: List<FunctionDeclaration
      */
     @JvmStatic
     public fun functionDeclarations(functionDeclarations: List<FunctionDeclaration>): Tool {
-      return Tool(functionDeclarations)
+      return Tool(functionDeclarations, null)
+    }
+
+    /**
+     * Creates a [Tool] instance that enables the model to use Grounding with Google Search.
+     *
+     * This allows the model to connect to Google Search to access and incorporate up-to-date
+     * information from the web into its responses.
+     *
+     * When this tool is used, the model's responses may include "Grounded Results" which are
+     * subject to the Grounding with Google Search terms outlined in the
+     * [Service Specific Terms](https://cloud.google.com/terms/service-terms).
+     *
+     * @param googleSearch An empty [GoogleSearch] object. The presence of this object in the list
```

Comment:
If it has to be an empty object, then it's the same object every time, then why we need it as param?

Reply:
sgtm